### ISSUE TITLE
PaDim without Ground Truth Labels + Using self-defined dataset

### ISSUE THREAD

**Issue by beaverCraze** [xiahaifeng1995/PaDiM-Anomaly-Detection-Localization-master#25]
Hi there,

I'm new to anomaly localization, and this implementation of PaDiM seems very friendly for solving my problem. Just one thing - I notice that PaDiM seems to require Ground Truth images (mask) of the training set. Is there a way to remove that? And also, how do I test the code on my custom dataset?

My dataset has 106 gray-scale rectangular images of dimension 4771x 3309. Any help is appreciated :)

**Comment by manureini** [comment]
No, it needs only ground truth images for testing.
I think you should create smaller patches for your image and you may need to learn a specific model for every patch.

**Comment by beaverCraze** [comment]
Thanks @manureini. Is there a way to remove the need of ground truth images for testing? Let's say I trained a model using PaDiM but want to test it on new unseen data - ideally, I don't want to make the ground truth version for each testing image, as this makes PaDiM semi-supervised. Any way to get around?

Lastly, for the smaller patches - do you mean cutting the image into equal sub-regions? Is there any recommendation on the dimensions of the sub-regions?

Thanks again 😃

**Comment by manureini** [comment]
Yes you can remove all evaluation code and it could work. But in general you want ground truth images, because I assume you want to know how well your model is working ;)
So in general you want to create some ground truth images.

Yes cutting the image in smaller regions. The images will be resized to 256x256 and cropped to 224x224 px. So anomalies should be visible in these patches.
